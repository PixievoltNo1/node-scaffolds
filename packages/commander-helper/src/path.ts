import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(
  workspace: string,
  targetPath?: string | null,
): string {
  if (targetPath == null) return workspace
  const absoluteDir: string = path.resolve(workspace, targetPath)
  return absoluteDir
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetDir
 */
export function relativeOfWorkspace(
  workspace: string,
  targetDir: string,
): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetDir)
  const relativeDir: string = path.relative(workspace, absoluteDir)
  return path.normalize(relativeDir)
}
