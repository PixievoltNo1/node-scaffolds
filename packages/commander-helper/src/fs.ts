import { isFileSync } from '@guanghechen/file-helper'
import fs from 'fs-extra'
import yaml from 'js-yaml'
import path from 'path'

/**
 * Load configuration file with format .json / .yml / .yaml
 *
 * @param filepath
 * @param logger
 */
export async function loadJsonOrYaml(
  filepath: string,
  encoding = 'utf-8',
): Promise<unknown | never> {
  if (!isFileSync(filepath))
    throw new Error(`${filepath} is an invalid file path`)

  const loadContent = (): Promise<string> =>
    fs.readFile(filepath, encoding as BufferEncoding)

  let result: unknown
  const extname = path.extname(filepath)
  switch (extname) {
    case '.json':
      {
        const content: string = await loadContent()
        result = JSON.parse(content)
      }
      break
    case '.yml':
    case '.yaml':
      {
        const content: string = await loadContent()
        result = yaml.load(content, { filename: filepath, json: true })
      }
      break
    default:
      throw new Error(
        `Only files in .json / .yml / .ymal format are supported. filepath(${filepath}`,
      )
  }
  return result
}

/**
 * Load configuration file with format .json / .yml / .yaml  (synchronizing)
 *
 * @param filepath
 * @param logger
 */
export function loadJsonOrYamlSync(
  filepath: string,
  encoding = 'utf-8',
): unknown | never {
  if (!isFileSync(filepath))
    throw new Error(`${filepath} is an invalid file path`)

  const loadContent = (): string =>
    fs.readFileSync(filepath, encoding as BufferEncoding)

  let result: unknown
  const extname = path.extname(filepath)
  switch (extname) {
    case '.json':
      {
        const content: string = loadContent()
        result = JSON.parse(content)
      }
      break
    case '.yml':
    case '.yaml':
      {
        const content: string = loadContent()
        result = yaml.load(content, { filename: filepath, json: true })
      }
      break
    default:
      throw new Error(
        `Only files in .json / .yml / .ymal format are supported. filepath(${filepath}`,
      )
  }
  return result
}
