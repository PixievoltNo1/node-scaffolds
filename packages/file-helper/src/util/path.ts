import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(workspace: string, targetPath?: string | null): string {
  if (targetPath == null) return workspace
  const filepath: string = path.resolve(workspace, targetPath)
  return path.normalize(filepath)
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetPath
 */
export function relativeOfWorkspace(workspace: string, targetPath: string): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetPath)
  const filepath: string = path.relative(workspace, absoluteDir)
  return path.normalize(filepath)
}
