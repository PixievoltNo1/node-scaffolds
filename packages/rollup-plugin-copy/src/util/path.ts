import path from 'path'

/**
 * Resolve the source path.
 * @param workspace
 * @param srcPath
 * @returns
 */
export const resolvePath = (workspace: string, srcPath: string): string => {
  const filepath = path.resolve(workspace, srcPath)
  return filepath
}

/**
 * Get the path relative to workspace.
 * @param workspace
 * @param filepath
 * @returns
 */
export const relativePath = (workspace: string, filepath: string): string => {
  const srcPath: string = filepath.startsWith(workspace)
    ? filepath.slice(workspace.length).replace(/^[/\\]/, '')
    : filepath
  return srcPath
}
