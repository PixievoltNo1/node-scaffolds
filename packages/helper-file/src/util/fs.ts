import fs from 'node:fs'
import path from 'node:path'

/**
 * Collect all files under the given directory.
 *
 * @param dir
 * @param predicate
 */
export async function collectAllFiles(
  dir: string,
  predicate?: ((p: string, stat: fs.Stats) => Promise<boolean> | boolean) | null,
): Promise<string[]> {
  const results: string[] = []
  await collect(dir)
  return results

  async function collect(filepath: string): Promise<void> {
    const stat = fs.statSync(filepath)
    if (stat.isDirectory()) {
      const filenames = fs.readdirSync(filepath)
      for (const filename of filenames) {
        const nextFilepath = path.join(filepath, filename)
        await collect(nextFilepath)
      }
      return
    }

    if (stat.isFile()) {
      if (predicate == null || (await predicate(filepath, stat))) {
        results.push(filepath)
      }
    }
  }
}

/**
 * Collect all files under the given directory.   (synchronizing)
 *
 * @param dir
 * @param predicate
 */
export function collectAllFilesSync(
  dir: string,
  predicate?: ((p: string, stat: fs.Stats) => boolean) | null,
): string[] {
  const results: string[] = []
  collect(dir)
  return results

  function collect(filepath: string): void {
    const stat = fs.statSync(filepath)
    if (stat.isDirectory()) {
      const filenames = fs.readdirSync(filepath)
      for (const filename of filenames) {
        const nextFilepath = path.join(filepath, filename)
        collect(nextFilepath)
      }
      return
    }
    if (stat.isFile()) {
      if (predicate == null || predicate(filepath, stat)) {
        results.push(filepath)
      }
    }
  }
}
