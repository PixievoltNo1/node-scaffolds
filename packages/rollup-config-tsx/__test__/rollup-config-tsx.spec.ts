import fs from 'fs-extra'
import path from 'path'
import { rollup } from 'rollup'
import type { OutputOptions, RollupOutput } from 'rollup'
import createRollupConfigs from '../src'

// Resolve absolute dirPath of case
const resolveCaseDir = (title: string): string =>
  path.resolve(__dirname, 'fixtures', title)

async function build(): Promise<RollupOutput[]> {
  // Create rollup config
  const { default: configOptions } = await import(path.resolve('config.ts'))
  const configs = createRollupConfigs(configOptions)

  const results: any[] = []
  for (const config of configs) {
    const bundle = await rollup(config)

    if (config.output == null) continue

    const outputOptions: OutputOptions[] = Array.isArray(config.output)
      ? config.output
      : [config.output]
    for (const outputOption of outputOptions) {
      const outputs = await bundle.generate(outputOption)
      for (const output of outputs.output as any[]) {
        const { format } = outputOption
        if (output.code != null) {
          results.push({ format, code: output.code })
        } else if (output.type === 'asset') {
          results.push({
            format,
            type: output.type,
            name: output.name,
            source: output.source.toString(),
            fileName: output.fileName,
          })
        } else {
          results.push({ format, ...output })
        }
      }
    }
  }
  return results
}

afterEach(async () => {
  await fs.remove('lib')
  await fs.remove('node_modules')
  process.chdir(__dirname)
})

describe('build', function () {
  // Timeout: 60s
  jest.setTimeout(60000)

  test('simple', async function () {
    const caseDir = resolveCaseDir('simple')
    process.chdir(caseDir)

    const results = await build()
    for (const result of results) {
      const { format, ...output } = result as any
      const data = output.code != null ? output.code : output
      expect(data).toMatchSnapshot(`rollup ${format}`)
    }
    // expect(fs.existsSync('lib/types/index.d.ts')).toBeTruthy()
    expect(fs.existsSync('lib/assets/font/tangerine.woff2')).toBeTruthy()
    expect(fs.existsSync('lib/assets/image/background.jpeg')).toBeTruthy()
    expect(fs.existsSync('src/style/index.styl.d.ts')).toBeTruthy()

    // Perform some cleanup operations.
    await fs.remove('src/style/index.styl.d.ts')
  })
})
