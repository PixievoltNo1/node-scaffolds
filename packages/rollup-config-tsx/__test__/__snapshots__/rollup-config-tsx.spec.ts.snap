// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build simple: rollup cjs 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cn = require('clsx');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cn__default = /*#__PURE__*/_interopDefaultLegacy(cn);

var name = \\"alice\\";
var data = {
  name: name
};

var classes = {\\"container\\":\\"ghc-container\\",\\"name\\":\\"ghc-name\\"};

function Greet(props) {
    var _a = props.name, name = _a === void 0 ? 'world' : _a;
    return (React__default[\\"default\\"].createElement(\\"div\\", { className: cn__default[\\"default\\"](classes.container) },
        \\"Hello, \\",
        React__default[\\"default\\"].createElement(\\"span\\", { className: cn__default[\\"default\\"](classes.name) }, name),
        \\"!\\"));
}

var index = (function () { return React__default[\\"default\\"].createElement(Greet, { name: data.name }); });

exports.Greet = Greet;
exports[\\"default\\"] = index;
"
`;

exports[`build simple: rollup cjs 2`] = `
Object {
  "fileName": "index.css",
  "name": undefined,
  "source": "/**
 * @see https://developers.google.com/fonts/docs/getting_started
 */
@font-face {
  font-family: 'Tangerine';
  font-style: normal;
  font-weight: 400;
  src: local('Tangerine Regular'), local('Tangerine-Regular'), url(\\"../assets/font/tangerine.woff2\\") format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}
.ghc-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 300px;
  height: 450px;
  margin: auto;
  background-image: url(\\"../assets/image/background.jpeg\\");
  background-size: contain;
  background-repeat: no-repeat;
}
.ghc-container .ghc-name {
  flex: 1 1;
  font-family: 'Tangerine', 'Comic Sans', serif;
  font-size: 48px;
  text-shadow: 4px 4px 4px #1a1a1a;
  color: #1a1a1a;
  padding: 1rem 2rem;
  white-space: nowrap;
}
.ghc-container .ghc-name ::-moz-placeholder {
  color: #808080;
}
.ghc-container .ghc-name ::placeholder {
  color: #808080;
}
",
  "type": "asset",
}
`;

exports[`build simple: rollup es 1`] = `
"import React from 'react';
import cn from 'clsx';

var name = \\"alice\\";
var data = {
  name: name
};

var classes = {\\"container\\":\\"ghc-container\\",\\"name\\":\\"ghc-name\\"};

function Greet(props) {
    var _a = props.name, name = _a === void 0 ? 'world' : _a;
    return (React.createElement(\\"div\\", { className: cn(classes.container) },
        \\"Hello, \\",
        React.createElement(\\"span\\", { className: cn(classes.name) }, name),
        \\"!\\"));
}

var index = (function () { return React.createElement(Greet, { name: data.name }); });

export { Greet, index as default };
"
`;

exports[`build simple: rollup es 2`] = `
Object {
  "fileName": "index.css",
  "name": undefined,
  "source": "/**
 * @see https://developers.google.com/fonts/docs/getting_started
 */
@font-face {
  font-family: 'Tangerine';
  font-style: normal;
  font-weight: 400;
  src: local('Tangerine Regular'), local('Tangerine-Regular'), url(\\"../assets/font/tangerine.woff2\\") format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}
.ghc-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 300px;
  height: 450px;
  margin: auto;
  background-image: url(\\"../assets/image/background.jpeg\\");
  background-size: contain;
  background-repeat: no-repeat;
}
.ghc-container .ghc-name {
  flex: 1 1;
  font-family: 'Tangerine', 'Comic Sans', serif;
  font-size: 48px;
  text-shadow: 4px 4px 4px #1a1a1a;
  color: #1a1a1a;
  padding: 1rem 2rem;
  white-space: nowrap;
}
.ghc-container .ghc-name ::-moz-placeholder {
  color: #808080;
}
.ghc-container .ghc-name ::placeholder {
  color: #808080;
}
",
  "type": "asset",
}
`;

exports[`build simple: rollup es 3`] = `
"import React from 'react';

/**
 * Params for constructing a Greet.
 */
interface GreetProps {
    name?: string;
}
declare function Greet(props: GreetProps): React.ReactElement;

declare const _default: () => JSX.Element;

export { Greet, GreetProps, _default as default };
"
`;

exports[`build simple: rollup undefined 1`] = `
"function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\"/**\\\\n * @see https://developers.google.com/fonts/docs/getting_started\\\\n */\\\\n@font-face {\\\\n  font-family: 'Tangerine';\\\\n  font-style: normal;\\\\n  font-weight: 400;\\\\n  src: local('Tangerine Regular'), local('Tangerine-Regular'), url(\\\\\\"../../lib/assets/font/tangerine.woff2\\\\\\") format('woff2');\\\\n  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;\\\\n}\\\\n.ghc-container {\\\\n  display: flex;\\\\n  align-items: center;\\\\n  justify-content: center;\\\\n  width: 300px;\\\\n  height: 450px;\\\\n  margin: auto;\\\\n  background-image: url(\\\\\\"../../lib/assets/image/background.jpeg\\\\\\");\\\\n  background-size: contain;\\\\n  background-repeat: no-repeat;\\\\n}\\\\n.ghc-container .ghc-name {\\\\n  flex: 1;\\\\n  font-family: 'Tangerine', 'Comic Sans', serif;\\\\n  font-size: 48px;\\\\n  text-shadow: 4px 4px 4px #1a1a1a;\\\\n  color: #1a1a1a;\\\\n  padding: 1rem 2rem;\\\\n  white-space: nowrap;\\\\n}\\\\n.ghc-container .ghc-name ::placeholder {\\\\n  color: #808080;\\\\n}\\\\n\\";
styleInject(css_248z);
"
`;
